main/src/ca/uwaterloo/flix/language/ast/Token.scala

```diff
@@ -15,4 +15,9 @@
  */
 package ca.uwaterloo.flix.language.ast
 
-case class Token(kind: TokenKind, text: String, line: Int, col: Int)
+case class Token(kind: TokenKind, src: Array[Char], startOffset: Int, endOffset: Int, line: Int, col: Int) {
+  /**
+   * Computes the lexeme that the token refers to by slicing it from `src`.
+   */
+  def text(): String = src.slice(startOffset, endOffset).mkString("")
```

Comment:
Perhaps this should be a lazy val and instead of slice and mkstring we can use substring?

Reply:
I made it into a `lazy val`. Didn't know about those, that's neat! 

`src` is an `Array[Char]` so I think we need to do slice + mkString.